parser/parser_manager.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package parser
+
+import (
+	"archive/tar"
+	"errors"
+	"io"
+)
+
+type ArtifactParser interface {
+	ParseHeader(tr *tar.Reader, hPath string) error
```

Comment:
Possibly, yes. But I want to make sure that you are providing both reader and writer to be used with this library.